proposals/core.md

```diff
@@ -147,38 +150,127 @@ WebDriver has a notion of a current top-level browsing context, and current brow
+    workers: DedicatedWorker[]
+    realm: Realm
+    // References to worklets
+    worklets: Worklet[]
+
+Realm
+    id: string
+    owner: BrowsingContext | Worker | Worklet
+
+Value
+    // Obtained from a Realm. Represents a JS value from that Realm.
+    // May represent an HTML Element.
+    realm: Realm
+    type: string
+    id: string?
+    value: object?
+```
 
-Today, WebDriver uses strings called "window handles" to uniquely identify top-level browsing contexts. These can easily be reused in the new protocol. Frames (nested browsing contexts) are a little tricker. There is no concept of a "frame ID" in WebDriver yet. The Switch To Frame command accepts either a number or a web element reference as a parameter. The number is context-sensitive and so it can't be used as a globally unique frame ID. Web element references are unique across all browsing contexts in a window so these are a better candidate. However, element references aren't valid across windows, so it seems we'll need to add a new type of globally unique ID for nested browsing contexts. These can be a simple string ID just like window handles. For interop purposes, we should also add commands for users to convert a browsing context ID to a web element reference that can be used with classical commands and vice-versa.
+The BiDi remote ends will send events to the client whenever new objects become available, or if an objects's basic properties (listed above) change.
 
-### Identifying windows and other top-level targets
+### Identifiers
 
-While we didn't arrive at any resolutions on the subject at TPAC this year, there was general interest in adding support for new contexts such as service workers and different JS realms. A WebDriver "window" in the current model considers only browsing contexts. It also conflates the concept of a browsing context with the concept of a script context. What this means in practice is that only document script contexts are visible to WebDriver. This is usually all the user needs/wants, but it precludes access to other script contexts such as web workers, service workers, and web extensions. Since some changes will already be necessary to allow the bidi protocol to target the contexts that already exists (i.e. frames), now seems as good a time as any to define these additional contexts and light up some new customer scenarios. Below is an updated browser model with some new (*) concepts:
+Every object in the model has an `id` property that is unique throughout the Session and indicates what type of object it is. An `id` is a human-readable type name, following by a string UUID. Examples:
 
-- WebDriver Session
-    - Target* (Formerly "Window". Can be a page or service worker)
-        - Browsing Context (aka "Frame")
-            - Element
-        - Script Context* (document, web worker, service worker, etc.)
+ - `"userAgent"` - The root user agent
+ - `"window/83d07169-4143-4ae8-947c-713f6949329f"` - A window target

```

Comment:
I"m not a fan of these complex string types that require parsing. Why not `{type: "window", "id": "83d07169-4143-4ae8-947c-713f6949329f"}` for example.